### PR TITLE
Drop support for 2020.3 and 2021.1 (closes #825)

### LINKED ISSUE #825: Drop support for 2021.1

### PR DESCRIPTION
**Pull request by mkondratek**

### CODE REVIEW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInBranchBelowAction.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ public void onFinished() {
         initialName,
         /* showCheckOutOption */ true,
         /* showResetOption */ true,
-        /* showSetTrackingOption */ false); // TODO (#496): use setTrackingOption that comes with 2020.2
+        /* showSetTrackingOption */ false);
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInBranchBelowAction.java:162]: that issue is closed and the comment is not applicable anyway

File: frontend/base/src/main/resources/GitMacheteBundle.properties [hunk 2]
```diff
@@ -46,7 +46,7 @@ action.GitMachete.BaseFastForwardMergeBranchToParentAction.description=Fast-forw
 action.GitMachete.BaseFastForwardMergeBranchToParentAction.notification.title.ff-fail=Fast-forward merge branch ''{1}'' to ''{0}'' failed.
 action.GitMachete.BaseFastForwardMergeBranchToParentAction.notification.title.ff-success.HTML=<html>Branch <b>{1}</b> has been fast-forward merged to <b>{0}</b></html>
 action.GitMachete.BaseFastForwardMergeBranchToParentAction.task-title=Fast-forward merging
-action.GitMachete.BaseFastForwardMergeBranchToParentAction.task-subtitle=Fast-forwarding...
+action.GitMachete.BaseFastForwardMergeBranchToParentAction.task-subtitle=Fast-forwarding\u2026
```
Comment (mkondratek) [inline frontend/base/src/main/resources/GitMacheteBundle.properties:49]: I ve checked button texts & progress task titles - it renders correctly ✅

File: frontend/base/src/test/java/com/virtuslab/gitmachete/resourcebundles/junit/GitMacheteBundlePropertiesTestSuite.java [hunk 3]
```diff
@@ -12,20 +13,39 @@ public class GitMacheteBundlePropertiesTestSuite {
...
+        .filter(prop -> ((String) prop.getValue()).contains("..."))
+        .forEach(t -> {
+          String key = (String) t.getKey();
+          fail("Properties should use ellipsis (\\u2026) instead of three dots (key=${key})");
+        });
+
+  }
+
+  private Properties getProperties() throws IOException {
+    Properties property = new Properties();
```
Comment (PawelLipski) [inline frontend/base/src/test/java/com/virtuslab/gitmachete/resourcebundles/junit/GitMacheteBundlePropertiesTestSuite.java:46]:
```suggestion
    Properties properties = new Properties();
```

File: CHANGE-NOTES.html [hunk 4]
```diff
@@ -1,3 +1 @@
-<p>The <i>Pull</i> action displays a clearer notification message when executed on branch that no longer exists in the remote repository.</p>
-
-<p>Stability fixes have been applied.</p>
+<p>Dropped support for IntelliJ 2021.1.</p>
```
Comment (PawelLipski) [inline CHANGE-NOTES.html:1]: And 2020.3 right? for clarity

**Comment by PawelLipski** [comment]
Another related TODO: https://github.com/VirtusLab/git-machete-intellij-plugin/pull/995#discussion_r924582231

**Comment by mkondratek** [comment]
part with moving icons is here tu

**Comment by PawelLipski** [comment]
> part with moving icons is here tu

Would be nice to also merge `frontend:icons` module to `frontend:base` (within the scope of #879, somewhat prematurely closed :/)

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -52,8 +52,8 @@ jetbrains-grammarkit = "org.jetbrains.grammarkit:2021.2.2"
 jetbrains-intellij = "org.jetbrains.intellij:1.7.0"
 # @pin because as per https://plugins.jetbrains.com/docs/intellij/kotlin.html#kotlin-standard-library:
 #      "If a plugin supports multiple platform versions, it must (...) target the lowest bundled stdlib version"
-#      which is 1.4.0 in our case (as this is the Kotlin version bundled into IntelliJ 2020.3).
-jetbrains-kotlin = "org.jetbrains.kotlin.jvm:1.4.0"
+#      which is different in our case (as this is the Kotlin version bundled into IntelliJ 2021.2).
+jetbrains-kotlin = "org.jetbrains.kotlin.jvm:1.5.10"
```
Comment (PawelLipski) [inline gradle/libs.versions.toml:56]: With this change, you can probably remove the `@pin` on `io.freefair.aspectj.post-compile-weaving`... pls check if the project still builds once that plugin is bumped to `6.5.0`

Comment (mkondratek) [reply gradle/libs.versions.toml]: i ve tried that. it does not work yet with kotlin jvm 1.5.10. I managed to get
image
and the already known https://github.com/freefair/gradle-plugins/issues/554 too

Comment (PawelLipski) [reply gradle/libs.versions.toml]: Oh! okay, so only works with Kotlin 1.6? did you already update on another PR?

Comment (mkondratek) [reply gradle/libs.versions.toml]: sadly it's neither working with 1.6 :/ I am getting the same errors there as here

Comment (PawelLipski) [reply gradle/libs.versions.toml]: Huh could you open a separate issue for that? 🤔

Comment (PawelLipski) [reply gradle/libs.versions.toml]: Maybe some non-trivial migration will be necessary. Plus update the comment on `@pin` accordingly

Comment (mkondratek) [reply gradle/libs.versions.toml]: I ve created an issue (https://github.com/freefair/gradle-plugins/issues/578).

Comment (mkondratek) [reply gradle/libs.versions.toml]: I spet some time on this one but I see no solution for now. Comment left and let's see what they say.

Comment (PawelLipski) [reply gradle/libs.versions.toml]: Whoops okay, so **not** a matter of Kotlin incompatibility this time ☹️

**Comment by PawelLipski** [comment]
Aaaand if you could also remove the instruction from Dockerfile: https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1064/files#diff-f34da55ca08f1a30591d8b0b3e885bcc678537b2a9a4aadea4f190806b374ddcR43

**Review by PawelLipski — APPROVED** [review]

File: docker/Dockerfile [hunk 6]
```diff
@@ -40,13 +40,6 @@ RUN \
   npm install --global remark-cli remark-lint-no-dead-urls remark-validate-links; \
   remark --help
 
-# TODO (#825): remove this `RUN`
```
Comment (PawelLipski) [inline docker/Dockerfile:43]: Pls also change image version in .circleci/config.yml to 8.0.0 (I've pushed it already)

**Pull request merged**